[Notifier] LOX24 SMS bridge (#54356)
symfony/symfony: src/Symfony/Component/Notifier/Bridge/Lox24/README.md

LOX24 SMS Notifier
==================

Provides [LOX24 SMS Gateway](https://doc.lox24.eu/#tag/sms/operation/api_sms_post_collection) integration for Symfony
Notifier.

DSN example
-----------

```

> LOX24_DSN=lox24://TOKEN@default?from=FROM&type=SERVICE_CODE&voice_language=VOICE_LANGUAGE&is_text_delete=IS_TEXT_DELETE&callback_data=CALLBACK_DATA

Reviewer: ```suggestion
LOX24_DSN=lox24://TOKEN@default?from=FROM&type=SERVICE_CODE&voice_language=VOICE_LANGUAGE&delete_text=DELETE_TEXT&callback_data=CALLBACK_DATA
```

to be more speakable

Reviewer: plus TOKEN Sems to be wrong, as we need user and token

```

where:

>  - `TOKEN` (required) is LOX24 API v2 token.

Reviewer: should be split as explained above

Author: fixed

 - `FROM` (required) is the sender of the message.
 - `TYPE` (optional) type of message: `sms` (by default) or `voice` (voice call).
 - `VOICE_LANGUAGE` (optional) if `type` is `voice`, then you can set the language of the voice message. Possible
  values: `de`, `en`, `es`, `fr`, `it` or `auto` (by default) per auto-detection.
 - `IS_TEXT_DELETE` (optional) delete SMS text from LOX24 database after sending SMS. Allowed values: `1` (true) or `0` (
  false). Default value: `0`.
 - `CALLBACK_DATA` (optional) additional data for the callback payload.

See your account info at https://account.lox24.eu

Send a Message
--------------

```php
use Symfony\Component\Notifier\Message\SmsMessage;

$sms = new SmsMessage('+1411111111', 'My message');

$texter->send($sms);
```

Advanced Message options
------------------------

```php
use Symfony\Component\Notifier\Message\SmsMessage;

> use Symfony\Component\Notifier\Bridge\LOX24\LOX24Options;

Reviewer: Please check the namespace across the bridge:
```diff
- LOX24
+ Lox24
```

> use Symfony\Component\Notifier\Bridge\LOX24\LOX24Options;

Reviewer: ```suggestion
use Symfony\Component\Notifier\Bridge\Lox24\Lox24Options;
```

$sms = new SmsMessage('+1411111111', 'My message');

> $options = (new LOX24Options())

Reviewer: ```suggestion
$options = (new Lox24Options())
```

    // set 'voice' per voice call (text-to-speech)
    ->type('voice')
    // set the language of the voice message.
    // If not set or set 'auto', the automatic language detection by message text will be used
    ->voiceLanguage('en')
    // unix timestamp of SMS the delivery time. If 0 or not set, the message will be sent immediately
    ->deliveryAt(new DateTime('2024-03-21 12:17:00'))
    // set True to delete the message from the LOX24 database after delivery
    ->textDelete(true)
    // pass any string to the callback object
    ->callbackData('some_data_per_callback');
    
// Add the custom options to the sms message and send the message
$sms->options($options);

$texter->send($sms);
```

Resources
---------

 * [Contributing](https://symfony.com/doc/current/contributing/index.html)
 * [Report issues](https://github.com/symfony/symfony/issues) and 

>  [send Pull Requests](https://github.com/symfony/symfony/pulls) 
>  in the [main Symfony repository](https://github.com/symfony/symfony)

Reviewer: ```suggestion
  [send Pull Requests](https://github.com/symfony/symfony/pulls) 
  in the [main Symfony repository](https://github.com/symfony/symfony)
```

>   [send Pull Requests](https://github.com/symfony/symfony/pulls) 
>   in the [main Symfony repository](https://github.com/symfony/symfony)

Reviewer: ```suggestion
   [send Pull Requests](https://github.com/symfony/symfony/pulls) 
   in the [main Symfony repository](https://github.com/symfony/symfony)
```

sorry, my fault, one more is needed

LOX24 SMS Notifier
==================

Provides [LOX24 SMS Gateway](https://doc.lox24.eu/#tag/sms/operation/api_sms_post_collection) integration for Symfony
Notifier.

DSN example
-----------

```

> LOX24_DSN=lox24://USER:TOKEN@default?from=FROM&type=SERVICE_CODE&voice_language=VOICE_LANGUAGE&delete_text=DELETE_TEXT&callback_data=CALLBACK_DATA

Reviewer 2: `voice_lang` no ?

Author: @smnandre that's bug. thanks! fixed!

> LOX24_DSN=lox24://USER:TOKEN@default?from=FROM&type=SERVICE_CODE&voice_language=VOICE_LANGUAGE&is_text_delete=DELETE_TEXT&callback_data=CALLBACK_DATA

Reviewer: Looks like this comment was lost somehow but please change:
```suggestion
LOX24_DSN=lox24://USER:TOKEN@default?from=FROM&type=SERVICE_CODE&voice_language=VOICE_LANGUAGE&delete_text=DELETE_TEXT&callback_data=CALLBACK_DATA
```

Author: fixed

Reviewer: delete_text please, it reads more natural

```

where:

 - `USER` (required) is LOX24 user ID.
 - `TOKEN` (required) is LOX24 API v2 token.
 - `FROM` (required) is the sender of the message.
 - `TYPE` (optional) type of message: `sms` (by default) or `voice` (voice call).
 - `VOICE_LANGUAGE` (optional) if `type` is `voice`, then you can set the language of the voice message. Possible
  values: `de`, `en`, `es`, `fr`, `it` or `auto` (by default) per auto-detection.

>  - `IS_TEXT_DELETE` (optional) delete SMS text from LOX24 database after sending SMS. Allowed values: `1` (true) or `0` (

Reviewer: ```suggestion
 - `DELETE_TEXT` (optional) delete SMS text from LOX24 database after sending SMS. Allowed values: `1` (true) or `0` (
```
```suggestion
 - `IS_TEXT_DELETE` (optional) delete SMS text from LOX24 database after sending SMS. Allowed values: `1` (true) or `0` (
```

Author: fixed

  false). Default value: `0`.
 - `CALLBACK_DATA` (optional) additional data for the callback payload.

See your account info at https://account.lox24.eu

Send a Message
--------------

```php
use Symfony\Component\Notifier\Message\SmsMessage;

$sms = new SmsMessage('+1411111111', 'My message');

$texter->send($sms);
```

Advanced Message options
------------------------

```php
use Symfony\Component\Notifier\Message\SmsMessage;
use Symfony\Component\Notifier\Bridge\Lox24\Lox24Options;

$sms = new SmsMessage('+1411111111', 'My message');

$options = (new Lox24Options())
    // set 'voice' per voice call (text-to-speech)
    ->type('voice')
    // set the language of the voice message.
    // If not set or set 'auto', the automatic language detection by message text will be used
    ->voiceLanguage('en')

>     // unix timestamp of SMS the delivery time. If 0 or not set, the message will be sent immediately

Reviewer 2: ```suggestion
    // Date of the SMS delivery. If null or not set, the message will be sent immediately
```

    ->deliveryAt(new DateTime('2024-03-21 12:17:00'))
    // set True to delete the message from the LOX24 database after delivery
    ->deleteTextAfterSending(true)
    // pass any string to the callback object
    ->callbackData('some_data_per_callback');
    
// Add the custom options to the sms message and send the message
$sms->options($options);

$texter->send($sms);
```

Resources
---------

 * [Contributing](https://symfony.com/doc/current/contributing/index.html)
 * [Report issues](https://github.com/symfony/symfony/issues) and 
   [send Pull Requests](https://github.com/symfony/symfony/pulls) 

>    in the [main Symfony repository](https://github.com/symfony/symfony)

Reviewer: New line please

Author: done
please add it to fabbot patches. really annoying

Reviewer: You can configure your IDE to do it for you all the time

Author: @OskarStark yes, I can. But why you set-up fabbot then? also, i think you tired to ask everybody to set-up their ide ;)

LOX24 SMS Notifier
=========================

Provides [LOX24 SMS Gateway](https://doc.lox24.eu/#tag/sms/operation/api_sms_post_collection) integration for Symfony Notifier.

DSN example
-----------

```
LOX24_DSN=lox24://TOKEN@default?from=FROM&type=SERVICE_CODE&voice_language=VOICE_LANGUAGE&is_text_delete=IS_TEXT_DELETE&callback_data=CALLBACK_DATA
```

where:

> - `TOKEN` (required) is LOX24 API v2 token.

Reviewer: ```suggestion
 - `TOKEN` (required) is LOX24 API v2 token.
```

Whitespace missing, same for the others

Author: done

- `FROM` (required) is the sender of the message.
- `TYPE` (optional) type of message: `sms` (by default) or `voice` (voice call).
- `VOICE_LANGUAGE` (optional) if `type` is `voice`, then you can set the language of the voice message. Possible values: `de`, `en`, `es`, `fr`, `it` or `auto` (by default) per auto-detection.
- `IS_TEXT_DELETE` (optional) delete SMS text from LOX24 database after sending SMS. Allowed values: `1` (true) or `0` (false). Default value: `0`.
- `CALLBACK_DATA` (optional) additional data for the callback payload.

See your account info at https://account.lox24.eu

Send a Message

> ---------------------------

Reviewer: Underline length must match the title length

Author: done

> With a LOX24 Message, you can use the `LOX24Options` class to add

Reviewer: I think this sentence needs to be removed

Author: done

message options.

```php
use Symfony\Component\Notifier\Message\SmsMessage;

$sms = new SmsMessage('+1411111111', 'My message');

$texter->send($sms);
```

Advanced Message options
---------------------------

With a LOX24 Message, you can use the `LOX24Options` class to add
message options.

```php
use Symfony\Component\Notifier\Message\SmsMessage;
use Symfony\Component\Notifier\Bridge\LOX24\LOX24Options;

$sms = new SmsMessage('+1411111111', 'My message');

$options = (new LOX24Options())
    // set 'voice' per voice call (text-to-speech)
    ->type('voice')
    // set the language of the voice message.
    // If not set or set 'auto', the automatic language detection by message text will be used
    ->voiceLanguage('en')
    // unix timestamp of SMS the delivery time. If 0 or not set, the message will be sent immediately

>     ->deliveryAt(1710871761)

Reviewer: I would prefer to have a DateTimeInterface for parameter signature instead of a Unix timestamp

Author: done

>     // set True per delete the message from the LOX24 database after delivery

Reviewer: ```suggestion
    // set True to delete the message from the LOX24 database after delivery
```

This explanation/comment should be used in the options class as well, it is much more understandable

Author: done

    ->textDelete(true)
    // pass any string to the callback object
    ->callbackData('some_data_per_callback');
    
// Add the custom options to the sms message and send the message
$sms->options($options);

$texter->send($sms);
```

Resources
---------

> * [Contributing](https://symfony.com/doc/current/contributing/index.html)
> * [Report issues](https://github.com/symfony/symfony/issues) and
>   [send Pull Requests](https://github.com/symfony/symfony/pulls)
>   in the [main Symfony repository](https://github.com/symfony/symfony)

Reviewer: ```suggestion
 * [Contributing](https://symfony.com/doc/current/contributing/index.html)
 * [Report issues](https://github.com/symfony/symfony/issues) and
   [send Pull Requests](https://github.com/symfony/symfony/pulls)
   in the [main Symfony repository](https://github.com/symfony/symfony)
```

Author: done
